data race in stoppable_waitgroup_test

```
WARNING: DATA RACE
Read at 0x00c0000d3500 by goroutine 26:
  math/rand.(*rngSource).Uint64()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rng.go:239 +0x3e
  math/rand.(*rngSource).Int63()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rng.go:234 +0x1d9
  math/rand.(*Rand).Int63()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:85 +0x85
  math/rand.(*Rand).Int31()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:99 +0x9a
  math/rand.(*Rand).Int31n()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:134 +0x4f
  math/rand.(*Rand).Intn()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:172 +0x59
  github.com/cilium/cilium/pkg/lock.(*SemaphoredMutexSuite).TestParallelism()
      /home/travis/gopath/src/github.com/cilium/cilium/pkg/lock/stoppable_waitgroup_test.go:203 +0x3a9
  runtime.call32()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/runtime/asm_amd64.s:540 +0x3d
  reflect.Value.Call()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/reflect/value.go:336 +0xd8
  gopkg.in/check%2ev1.(*suiteRunner).forkTest.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:781 +0xabb
  gopkg.in/check%2ev1.(*suiteRunner).forkCall.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:675 +0xe1
Previous write at 0x00c0000d3500 by goroutine 27:
  math/rand.(*rngSource).Uint64()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rng.go:239 +0x54
  math/rand.(*rngSource).Int63()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rng.go:234 +0x1d9
  math/rand.(*Rand).Int63()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:85 +0x125
  math/rand.(*Rand).Int31()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:99 +0x135
  math/rand.(*Rand).Int31n()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:131 +0x130
  math/rand.(*Rand).Intn()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:172 +0x59
  github.com/cilium/cilium/pkg/lock.(*SemaphoredMutexSuite).TestParallelism.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/pkg/lock/stoppable_waitgroup_test.go:178 +0x64
Goroutine 26 (running) created at:
  gopkg.in/check%2ev1.(*suiteRunner).forkCall()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:672 +0x459
  gopkg.in/check%2ev1.(*suiteRunner).forkTest()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:763 +0x1c4
  gopkg.in/check%2ev1.(*suiteRunner).runTest()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:818 +0x23e
  gopkg.in/check%2ev1.(*suiteRunner).run()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:624 +0x1dc
  gopkg.in/check%2ev1.Run()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/run.go:92 +0x5a
  gopkg.in/check%2ev1.RunAll()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/run.go:84 +0x136
  gopkg.in/check%2ev1.TestingT()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/run.go:72 +0x5ef
  github.com/cilium/cilium/pkg/lock.Test()
      /home/travis/gopath/src/github.com/cilium/cilium/pkg/lock/lock_test.go:27 +0x38
  testing.tRunner()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/testing/testing.go:1127 +0x202
Goroutine 27 (running) created at:
  github.com/cilium/cilium/pkg/lock.(*SemaphoredMutexSuite).TestParallelism()
      /home/travis/gopath/src/github.com/cilium/cilium/pkg/lock/stoppable_waitgroup_test.go:175 +0x2a5
  runtime.call32()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/runtime/asm_amd64.s:540 +0x3d
  reflect.Value.Call()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/reflect/value.go:336 +0xd8
  gopkg.in/check%2ev1.(*suiteRunner).forkTest.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:781 +0xabb
  gopkg.in/check%2ev1.(*suiteRunner).forkCall.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:675 +0xe1
==================
==================
WARNING: DATA RACE
Read at 0x00c0000d3508 by goroutine 26:
  math/rand.(*rngSource).Uint64()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rng.go:244 +0x8e
  math/rand.(*rngSource).Int63()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rng.go:234 +0x1d9
  math/rand.(*Rand).Int63()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:85 +0x85
  math/rand.(*Rand).Int31()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:99 +0x9a
  math/rand.(*Rand).Int31n()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:134 +0x4f
  math/rand.(*Rand).Intn()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:172 +0x59
  github.com/cilium/cilium/pkg/lock.(*SemaphoredMutexSuite).TestParallelism()
      /home/travis/gopath/src/github.com/cilium/cilium/pkg/lock/stoppable_waitgroup_test.go:203 +0x3a9
  runtime.call32()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/runtime/asm_amd64.s:540 +0x3d
  reflect.Value.Call()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/reflect/value.go:336 +0xd8
  gopkg.in/check%2ev1.(*suiteRunner).forkTest.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:781 +0xabb
  gopkg.in/check%2ev1.(*suiteRunner).forkCall.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:675 +0xe1
Previous write at 0x00c0000d3508 by goroutine 27:
  math/rand.(*rngSource).Uint64()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rng.go:244 +0xaa
  math/rand.(*rngSource).Int63()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rng.go:234 +0x1d9
  math/rand.(*Rand).Int63()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:85 +0x125
  math/rand.(*Rand).Int31()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:99 +0x135
  math/rand.(*Rand).Int31n()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:131 +0x130
  math/rand.(*Rand).Intn()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:172 +0x59
  github.com/cilium/cilium/pkg/lock.(*SemaphoredMutexSuite).TestParallelism.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/pkg/lock/stoppable_waitgroup_test.go:178 +0x64
Goroutine 26 (running) created at:
  gopkg.in/check%2ev1.(*suiteRunner).forkCall()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:672 +0x459
  gopkg.in/check%2ev1.(*suiteRunner).forkTest()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:763 +0x1c4
  gopkg.in/check%2ev1.(*suiteRunner).runTest()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:818 +0x23e
  gopkg.in/check%2ev1.(*suiteRunner).run()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:624 +0x1dc
  gopkg.in/check%2ev1.Run()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/run.go:92 +0x5a
  gopkg.in/check%2ev1.RunAll()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/run.go:84 +0x136
  gopkg.in/check%2ev1.TestingT()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/run.go:72 +0x5ef
  github.com/cilium/cilium/pkg/lock.Test()
      /home/travis/gopath/src/github.com/cilium/cilium/pkg/lock/lock_test.go:27 +0x38
  testing.tRunner()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/testing/testing.go:1127 +0x202
Goroutine 27 (running) created at:
  github.com/cilium/cilium/pkg/lock.(*SemaphoredMutexSuite).TestParallelism()
      /home/travis/gopath/src/github.com/cilium/cilium/pkg/lock/stoppable_waitgroup_test.go:175 +0x2a5
  runtime.call32()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/runtime/asm_amd64.s:540 +0x3d
  reflect.Value.Call()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/reflect/value.go:336 +0xd8
  gopkg.in/check%2ev1.(*suiteRunner).forkTest.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:781 +0xabb
  gopkg.in/check%2ev1.(*suiteRunner).forkCall.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:675 +0xe1
==================
==================
WARNING: DATA RACE
Read at 0x00c0000d3638 by goroutine 26:
  math/rand.(*rngSource).Uint64()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rng.go:249 +0x164
  math/rand.(*rngSource).Int63()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rng.go:234 +0x1d9
  math/rand.(*Rand).Int63()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:85 +0x85
  math/rand.(*Rand).Int31()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:99 +0x9a
  math/rand.(*Rand).Int31n()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:134 +0x4f
  math/rand.(*Rand).Intn()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:172 +0x59
      /home/travis/gopath/src/github.com/cilium/cilium/pkg/lock/stoppable_waitgroup_test.go:203 +0x3a9
  runtime.call32()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/runtime/asm_amd64.s:540 +0x3d
  reflect.Value.Call()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/reflect/value.go:336 +0xd8
  gopkg.in/check%2ev1.(*suiteRunner).forkTest.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:781 +0xabb
  gopkg.in/check%2ev1.(*suiteRunner).forkCall.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:675 +0xe1
Previous write at 0x00c0000d3638 by goroutine 27:
  math/rand.(*rngSource).Uint64()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rng.go:250 +0x1c4
  math/rand.(*rngSource).Int63()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rng.go:234 +0x1d9
  math/rand.(*Rand).Int63()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:85 +0x125
  math/rand.(*Rand).Int31()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:99 +0x135
  math/rand.(*Rand).Int31n()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:131 +0x130
  math/rand.(*Rand).Intn()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:172 +0x59
  github.com/cilium/cilium/pkg/lock.(*SemaphoredMutexSuite).TestParallelism.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/pkg/lock/stoppable_waitgroup_test.go:178 +0x64
Goroutine 26 (running) created at:
  gopkg.in/check%2ev1.(*suiteRunner).forkCall()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:672 +0x459
  gopkg.in/check%2ev1.(*suiteRunner).forkTest()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:763 +0x1c4
  gopkg.in/check%2ev1.(*suiteRunner).runTest()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:818 +0x23e
  gopkg.in/check%2ev1.(*suiteRunner).run()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:624 +0x1dc
  gopkg.in/check%2ev1.Run()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/run.go:92 +0x5a
  gopkg.in/check%2ev1.RunAll()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/run.go:84 +0x136
  gopkg.in/check%2ev1.TestingT()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/run.go:72 +0x5ef
  github.com/cilium/cilium/pkg/lock.Test()
      /home/travis/gopath/src/github.com/cilium/cilium/pkg/lock/lock_test.go:27 +0x38
  testing.tRunner()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/testing/testing.go:1127 +0x202
Goroutine 27 (running) created at:
  github.com/cilium/cilium/pkg/lock.(*SemaphoredMutexSuite).TestParallelism()
      /home/travis/gopath/src/github.com/cilium/cilium/pkg/lock/stoppable_waitgroup_test.go:175 +0x2a5
  runtime.call32()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/runtime/asm_amd64.s:540 +0x3d
  reflect.Value.Call()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/reflect/value.go:336 +0xd8
  gopkg.in/check%2ev1.(*suiteRunner).forkTest.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:781 +0xabb
  gopkg.in/check%2ev1.(*suiteRunner).forkCall.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:675 +0xe1
==================
==================
WARNING: DATA RACE
Write at 0x00c0000d40a8 by goroutine 26:
  math/rand.(*rngSource).Uint64()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rng.go:250 +0x1c4
  math/rand.(*rngSource).Int63()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rng.go:234 +0x1d9
  math/rand.(*Rand).Int63()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:85 +0x85
  math/rand.(*Rand).Int31()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:99 +0x9a
  math/rand.(*Rand).Int31n()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:134 +0x4f
  math/rand.(*Rand).Intn()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:172 +0x59
  github.com/cilium/cilium/pkg/lock.(*SemaphoredMutexSuite).TestParallelism()
      /home/travis/gopath/src/github.com/cilium/cilium/pkg/lock/stoppable_waitgroup_test.go:203 +0x3a9
  runtime.call32()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/runtime/asm_amd64.s:540 +0x3d
  reflect.Value.Call()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/reflect/value.go:336 +0xd8
  gopkg.in/check%2ev1.(*suiteRunner).forkTest.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:781 +0xabb
  gopkg.in/check%2ev1.(*suiteRunner).forkCall.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:675 +0xe1
Previous read at 0x00c0000d40a8 by goroutine 27:
  math/rand.(*rngSource).Uint64()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rng.go:249 +0x164
  math/rand.(*rngSource).Int63()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rng.go:234 +0x1d9
  math/rand.(*Rand).Int63()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:85 +0x125
  math/rand.(*Rand).Int31()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:99 +0x135
  math/rand.(*Rand).Int31n()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:131 +0x130
  math/rand.(*Rand).Intn()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/math/rand/rand.go:172 +0x59
  github.com/cilium/cilium/pkg/lock.(*SemaphoredMutexSuite).TestParallelism.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/pkg/lock/stoppable_waitgroup_test.go:178 +0x64
Goroutine 26 (running) created at:
  gopkg.in/check%2ev1.(*suiteRunner).forkCall()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:672 +0x459
  gopkg.in/check%2ev1.(*suiteRunner).forkTest()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:763 +0x1c4
  gopkg.in/check%2ev1.(*suiteRunner).runTest()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:818 +0x23e
  gopkg.in/check%2ev1.(*suiteRunner).run()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:624 +0x1dc
  gopkg.in/check%2ev1.Run()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/run.go:92 +0x5a
  gopkg.in/check%2ev1.RunAll()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/run.go:84 +0x136
  gopkg.in/check%2ev1.TestingT()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/run.go:72 +0x5ef
  github.com/cilium/cilium/pkg/lock.Test()
      /home/travis/gopath/src/github.com/cilium/cilium/pkg/lock/lock_test.go:27 +0x38
  testing.tRunner()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/testing/testing.go:1127 +0x202
Goroutine 27 (running) created at:
  github.com/cilium/cilium/pkg/lock.(*SemaphoredMutexSuite).TestParallelism()
      /home/travis/gopath/src/github.com/cilium/cilium/pkg/lock/stoppable_waitgroup_test.go:175 +0x2a5
  runtime.call32()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/runtime/asm_amd64.s:540 +0x3d
  reflect.Value.Call()
      /home/travis/.gimme/versions/go1.15.2.linux.amd64/src/reflect/value.go:336 +0xd8
  gopkg.in/check%2ev1.(*suiteRunner).forkTest.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:781 +0xabb
  gopkg.in/check%2ev1.(*suiteRunner).forkCall.func1()
      /home/travis/gopath/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:675 +0xe1
==================
OK: 9 passed
    testing.go:1042: race detected during execution of test
--- FAIL: Test (2.42s)
=== CONT  
    testing.go:1042: race detected during execution of test
```

--- a/pkg/lock/stoppable_waitgroup_test.go
+++ b/pkg/lock/stoppable_waitgroup_test.go
@@ -167,15 +167,16 @@ func (s *SemaphoredMutexSuite) TestWaitChannel(c *C) {
 func (s *SemaphoredMutexSuite) TestParallelism(c *C) {
 	l := NewStoppableWaitGroup()
 
-	// use math/rand instead of pkg/rand to avoid a test import cycle which
-	// go vet would complain about.
-	randGen := rand.New(rand.NewSource(time.Now().UnixNano()))
+	// Use math/rand instead of pkg/rand to avoid a test import cycle which
+	// go vet would complain about. Use the global default entropy source
+	// rather than creating a new source to avoid concurrency issues.
+	rand.Seed(time.Now().UnixNano())
 	in := make(chan int)
 	stop := make(chan struct{})
 	go func() {
 		for {
 			select {
-			case in <- randGen.Intn(1 - 0):
+			case in <- rand.Intn(1 - 0):
 			case <-stop:
 				close(in)
 				return
@@ -200,7 +201,7 @@ func (s *SemaphoredMutexSuite) TestParallelism(c *C) {
 		}()
 	}
 
-	time.Sleep(time.Duration(randGen.Intn(3-0)) * time.Second)
+	time.Sleep(time.Duration(rand.Intn(3-0)) * time.Second)
 	close(stop)
 	wg.Wait()
 	add := atomic.LoadInt64(&adds)

